src/controllers/accommodationController.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable max-len */
+import Model from '../models';
+import * as accommodationServices from '../services/accommodationServices';
+import { respondWithSuccess, respondWithWarning } from '../helpers/responseHandler';
+
+const { Accommodation, Room } = Model;
+/**
+ * Function creates accommodation
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} response object
+ */
+export const createAccommodation = async (req, res) => {
+  try {
+    req.accommodation = req.body;
+    req.accommodation.userId = req.auth.id;
+    req.accommodation.images = req.files.map(file => file.secure_url);
+    const accommodation = await accommodationServices.create(Accommodation, req.accommodation);
+    return respondWithSuccess(res, 201, 'resource successfully created', accommodation.dataValues);
+  } catch (error) {
+    return respondWithWarning(res, 500, 'Server Error');
+  }
+};
+
+/**
+ * Function creates rooms for an accommodation
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} response object
+ */
+export const createRoom = async (req, res) => {
+  try {
+    req.room = req.body;
+    req.room.accommodationId = parseInt(req.params.accommodationId, 10);
```

Comment:
same here. Let's try to avoid adding properties to the `req` object if we're not going to pass to the next middleware or controller.